eth/api.go

```diff
@@ -438,10 +441,15 @@ func (s *PrivateAccountAPI) NewAccount(password string) (common.Address, error)
 	return common.Address{}, err
 }
 
-// UnlockAccount will unlock the account associated with the given address with the given password for duration seconds.
-// It returns an indication if the action was successful.
-func (s *PrivateAccountAPI) UnlockAccount(addr common.Address, password string, duration int) bool {
-	if err := s.am.TimedUnlock(addr, password, time.Duration(duration)*time.Second); err != nil {
+// UnlockAccount will unlock the account associated with the given address with
+// the given password for duration seconds. If duration is nil it will use a
+// default of 300 seconds. It returns an indication if the account was unlocked.
+func (s *PrivateAccountAPI) UnlockAccount(addr common.Address, password string, duration *rpc.HexNumber) bool {
+	if duration == nil {
+		duration = rpc.NewHexNumber(300)
+	}
```

Comment:
I'm not sure about this decision. If I unlock the account and don't specify a timeout, I expect it to be unlocked indefinitely. Adding this timeout here makes it impossible to indefinitely unlock an account. Since we provide an account lock counterpart, I think we can leave it up to the dapp to handle the account properly and try to not be too smart. But I'm open to hear why this is a better idea this way perhaps.